samcli/cli/command.py

```diff
@@ -110,3 +111,28 @@ def get_command(self, ctx, cmd_name):
             return
 
         return mod.cli
+
+    # NOTE(TheSriram): info_name is always set to "sam". This way when the CLI is invoked as a module,
+    # the help text that is generated still says "sam" instead of "__main__".
+    def make_context(self, info_name, args, parent=None, **extra):
```

Comment:
Original make_context function: https://github.com/pallets/click/blob/38a2712d596a5df9288c1817748b71197d8c7c57/click/core.py#L621

Reply:
https://github.com/pallets/click/issues/82  is a better way to do it.

Reply:
That is super clean. Nice find!